bpf/flows.c

```diff
@@ -55,16 +52,47 @@
  */
 #include "pkt_translation.h"
 
-static inline void update_existing_flow(flow_metrics *aggregate_flow, pkt_info *pkt, u64 len,
-                                        u32 sampling) {
-    bpf_spin_lock(&aggregate_flow->lock);
-    aggregate_flow->packets += 1;
-    aggregate_flow->bytes += len;
-    aggregate_flow->end_mono_time_ts = pkt->current_ts;
-    aggregate_flow->flags |= pkt->flags;
-    aggregate_flow->dscp = pkt->dscp;
-    aggregate_flow->sampling = sampling;
-    bpf_spin_unlock(&aggregate_flow->lock);
+static inline void add_observed_intf(flow_metrics *value, pkt_info *pkt, u32 if_index,
+                                     u8 direction) {
+    u8 nb_observed_intf = value->nb_observed_intf;
+    if (nb_observed_intf < MAX_OBSERVED_INTERFACES) {
+        for (u8 i = 0; i < nb_observed_intf; i++) {
+            if (value->observed_intf[i] == if_index) {
+                // Interface already seen -> skip (we don't check for direction to avoid reaching the max too frequently, as a tradeoff)
```

Comment:
what is the impact of ignoring direction ?

Reply:
We discussed that in slack. The result is not marking twice the same interface when it has a different direction. E.g. if a flow shows eth0/egress then eth0/ingress, only eth0/egress is going to be stored.
We may document that somewhere as a limitation ...

Reply:
as said in slack an alternative could be to introduce a new direction enum "BOTH", if we think it's important. It would take both of both worlds, ie. avoid reaching the max interface and still telling everything about what was observed. But it needs changes in the console plugin for display. I can do that as a follow-up if we think it's important

Reply:
It's mostly theoretical, I don't think it involve any real loss in user experience - in fact, when I was seeing this the other day, I'm now pretty sure it was due to the non-IP flows that we're [not capturing anymore](https://github.com/netobserv/netobserv-ebpf-agent/pull/509/commits/938e96dce3cc3ab1b654a0c0aa855228b99d15eb).

Anyway, it's not a big deal to manage "both" directions so I've added that in the last commit. And on console side: https://github.com/netobserv/network-observability-console-plugin/pull/695 (I generated fake data to be able to see it in console)